lua/lv-utils/init.lua

```diff
@@ -1,5 +1,61 @@
 local lv_utils = {}
 
+-- recursive Print (structure, limit, indent)
+local function rPrint(s, l, i)
+  l = l or 100 -- default item limit
+  i = i or "." -- indent string
+  if l < 1 then
+    print "ERROR: Item limit reached."
+    return l - 1
+  end
+  local ts = type(s)
+  if ts ~= "table" then
+    if ts == nil then
+      io.write("-- O", i:sub(2), " = nil\n")
+      return l - 1
+    elseif ts == "string" then
+      -- escape sequences
+      s = string.format("%q", s)
+    end
+    i = i:gsub("%.%[", "%[")
+    if type(s) == "function" then
+      -- don't print functions
+      io.write("-- O", i:sub(2), " = function ()\n")
+    else
+      io.write("O", i:sub(2), " = ", tostring(s), "\n")
+    end
+    return l - 1
+  end
+
+  if ts == nil then
+    ts = ""
+  end
+  for k, v in pairs(s) do
```

Comment:
You could probably put this in a `if ts == 'table'`condition on top and return from there (the motivation is that it contains the recursion) and as a consequence you would be able to unwrap the other type checks from the `if ts ~= "table"` condition.